tests/test_compat.py

```diff
@@ -0,0 +1,46 @@
+# compat.py
+# Python (2 -v- 3) compatibility functions.
+#
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU Lesser General Public License v.2, or (at your option) any later
+# version. This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY expressed or implied, including the implied
+# warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+# the GNU Lesser General Public License for more details.  You should have
+# received a copy of the GNU Lesser General Public License along with this
+# program; if not, write to the Free Software Foundation, Inc., 51 Franklin
+# Street, Fifth Floor, Boston, MA 02110-1301, USA.  Any Red Hat trademarks
+# that are incorporated in the source code or documentation are not subject
+# to the GNU Lesser General Public License and may only be used or
+# replicated with the express permission of Red Hat, Inc.
+#
+# Red Hat Author(s): David Lehman <dlehman@redhat.com>
+#
+
+import six as _six
+
+mock_move = _six.MovedModule('mock', 'mock', 'unittest.mock')
+has_mock = False
+
+
+def add_move(mod):
+    _six.add_move(mod)
+    # https://bitbucket.org/gutworth/six/issues/116/enable-importing-from-within-custom
+    _six._importer._add_module(mod, "moves." + mod.name)
+
+
+def setup():
+    global has_mock
+    add_move(mock_move)
+
+    try:
+        import six.moves.mock as _mock
+    except ImportError:
+        has_mock = False
+    else:
+        has_mock = True
```

Comment:
Well, we don't want the `ImportError` to cause the test run to end -- what we want is to adapt by only running the tests that do not require `mock`. Does that seem reasonable?